lib/puppet/defaults.rb

```diff
@@ -1395,7 +1395,24 @@ def self.default_diffargs
     },
     :server => {
       :default => "puppet",
-      :desc => "The puppet master server to which the puppet agent should connect."
+      :desc => "The puppet master server to which the puppet agent should connect.",
+      :hook => proc { |value|
+          if Puppet.settings.set_by_config?(:server_list)
```

Comment:
deprecating `server` is going to effect many code paths... Can we either not deprecate it and accept both (with one taking priority) or just change `server` to be multivalued, so that existing code doesn't need to change and will continue to do the right thing?